src/practices/CPP/CPPLinterUsedPractice.ts

```diff
@@ -0,0 +1,29 @@
+import { PracticeContext } from '../../contexts/practice/PracticeContext';
+import { PracticeEvaluationResult, PracticeImpact, ProgrammingLanguage } from '../../model';
+import { DxPractice } from '../DxPracticeDecorator';
+import { IPractice } from '../IPractice';
+
+@DxPractice({
+  id: 'CPP.LinterUsedPractice',
+  name: 'Use a C++ Linter',
+  impact: PracticeImpact.medium,
+  suggestion: 'Use Linter to catch bugs and dangerous code constructs.',
+  reportOnlyOnce: true,
+  url: 'https://github.com/caramelomartins/awesome-linters#cc',
+})
+export class CPPLinterUsedPractice implements IPractice {
+  async isApplicable(ctx: PracticeContext): Promise<boolean> {
+    return ctx.projectComponent.language === ProgrammingLanguage.CPlusPlus;
+  }
+
+  async evaluate(ctx: PracticeContext): Promise<PracticeEvaluationResult> {
+    if (ctx.packageInspector) {
+      if (ctx.packageInspector.hasOneOfPackages(['cppcheck.sourceforge.net/', 'github.com/oclint/oclint'])) {
```

Comment:
I have the opportunity to discuss this with C++ developer. He said he doesn't seem the possibility to check using C++ libraries like this since there is no package manager. Are you a C++ developer? Do you know how this is working? I'm not telling it's impossible but please, if you can describe me how this should work, it would be great. Thanks!

Reply:
No, I am not a C++ developer.
![image](https://user-images.githubusercontent.com/42507632/96746729-1d016000-13e5-11eb-9264-2ba4edd09046.png)
Actually for any repository dealing with C++ codebase, if these packages are present in the repository, then it can be used to do linting and related checks like this:-
![image](https://user-images.githubusercontent.com/42507632/96747048-75386200-13e5-11eb-81ca-0cec1e603f05.png)

But Ma'am I am not sure of how this package will be checked like the link directs to the official page about the tool, which I also saw in `GoLinterUsedPractises.ts`
🤔

Reply:
You can't compare two different languages. Go can have the packages in the `go.mod` file. See the [GoPackageInspector](https://github.com/DXHeroes/dx-scanner/blob/master/src/inspectors/package/GoPackageInspector.ts). In NPM there are two package managers - npm and yarn. So it's easy to get the name of used packages (if they are installed locally for the project, no globally). 
See the [JavaScriptPackageInspector](https://github.com/DXHeroes/dx-scanner/blob/master/src/inspectors/package/JavaScriptPackageInspector.ts)
Or any other package inspector in [package folder](https://github.com/DXHeroes/dx-scanner/tree/master/src/inspectors/package). 
There are some package managers in C++ as I found by googling, but they are not widely used. You can implement the practice, but I'm not sure it'll be working after someone add the support for the C++...

The reason why we don't support some of the languages is, that we are not developers e.g. in C++. So we taught that if someone chooses some practice for other languages than JS/TS, he knows the language.